### ISSUE TITLE
Websocket fails under load with error message when sending to client

### ISSUE THREAD

**Issue by mvkra** [SuaveIO/suave#467]
Another bug around websockets unfortunately. After a minute or two sending large payloads over the websocket I'm getting the below exception when Suave is trying to send to the client. After looking at the Suave code here my suspicion is that the same SocketAsyncEventArgs (the one TcpTransport is initialised with) is being used concurrently by two async threads. This is probably because one isn't finished sending its data when another websocket send is triggered by my application code - the websocket code uses the ITransport.write member. Happy to provide additional information of course.

```
Exception: 
[System.InvalidOperationException] = {"\"An asynchronous socket operation is already in progress using this SocketAsyncEventArgs instance.\";"}

Stack Trace: 
   at System.Net.Sockets.SocketAsyncEventArgs.StartConfiguring()
   at System.Net.Sockets.SocketAsyncEventArgs.SetBufferInternal(Byte[] buffer, Int32 offset, Int32 count)
   at <StartupCode$Suave>.$TcpTransport.Suave-Sockets-ITransport-write@29-1.Invoke(SocketAsyncEventArgs args)
   at Suave.Sockets.Utils.asyncDo@119.Invoke(Tuple`3 tupledArg)
   at <StartupCode$FSharp-Core>.$Control.FromContinuations@1287.Invoke(AsyncParams`1 _arg1)
```

Some idea of what size data I'm sending and how often:
Time	        Size in Bytes	Size in MB
14:42:36	30	                2.86102E-05
14:42:39	2385805	        2.275280952
14:42:42	244568	        0.23323822
14:42:47	12124898	11.56320381
14:42:59	89217914	85.08483315
14:43:06	67835098	64.69259071
14:43:11	41365746	39.44944954
14:43:17	58782054	56.05893517
14:43:18	11743062  	11.19905663
14:43:26	34336306	32.74565315
14:43:26	42744394	40.76423073
14:43:35	10093778	9.626176834
14:43:35	24102318	22.98576164
14:43:40	5161284	        4.92218399
14:43:40	5639958	        5.37868309
14:43:43	19607828	18.69948196

**Comment by ademar** [comment]
@mvkra are you running the latest pre-release 2.0.0-alpha4 ?

**Comment by mvkra** [comment]
@ademar I just tested it under the 2.0.0-alpha4 pre-release package and yes the issue still persists.

**Comment by ademar** [comment]
Would you mind sharing your failing POC so we do not have to cook it ourselves ? I also was wondering what platform are you on ?

Thanks.

**Comment by mvkra** [comment]
On Windows 7 (standard .NET framework 4.5). When I get the time will try to create a POC.

**Comment by ademar** [comment]
Hum ... at some point had a gate right there. It seems it fell through the cracks. 

Let me research a little bit and will bring it back.

**Comment by mvkra** [comment]
No problem - I've forked the Suave repo with and added the code I used to remove my issue if its helpful. This is a bit of a showstopper for my usage so obviously would like to see this fixed and put out soon. Note that you needed a lock on the setBuffer call as well or anything that interacts with the socket.

**Comment by ademar** [comment]
@mvkra Can you give a try to the code in the PR #470

**Comment by mvkra** [comment]
@ademar After testing this for about 20 minutes with the new library I didn't get a crash. The current version failed in about a minute. So your changes I think have solved the issue. Just wondering whether a bugfix release of 1.1.x can be released rather than waiting for 2.0.0 (stable) to be ready which might take awhile.

**Comment by ademar** [comment]
@mvkra I've released v1.1.3 with the fix. Let us know if you encounter any issues.

**Comment by mvkra** [comment]
All good with the new version on my end. Will close the issue.
